scripts/operations/configuration/write_root_secrets_to_vault.py

```diff
@@ -76,38 +85,138 @@ def get_root_hash_from_etc_shadow() -> str:
+    variables. This will require passwords to be at least 8 characters long.
     """
+    logging.info(
+        f"Reading in plaintext password from {env_var_name} environment variable")
+    return args.get_env_var_value(
+        env_var_name=env_var_name,
+        value_validator=lambda s: args.validate_string(s, min_length=8))
 
-    # Read in secrets from the system
-    try:
-        ssh_private_key = common.read_file("/root/.ssh/id_rsa")
-        ssh_public_key = common.read_file("/root/.ssh/id_rsa.pub")
-        password_hash = get_root_hash_from_etc_shadow()
-        vault.write_root_secrets(ssh_private_key=ssh_private_key,
-                                 ssh_public_key=ssh_public_key,
-                                 password_hash=password_hash,
-                                 verify=True)
-    except common.ScriptException as e:
-        common.print_err_exit(f"{e}")
 
-    print("SUCCESS", flush=True)
+def pw_hash_env_var(env_var_name: str) -> str:
+    """
+    Wrapper for args.get_env_var_value with appropriate arguments for password hash
+    environment variables.
+
+    The minimum length of the password hash will depend on the algorithm that is used.
+    For our purposes, we will look for an 8 character minimum, and make sure that it
+    begins with a $ character.
+    """
+    logging.info(
+        f"Reading in password hash from {env_var_name} environment variable")
+    return args.get_env_var_value(
+        env_var_name=env_var_name,
+        value_validator=lambda s: args.validate_string(s, min_length=8, required_prefix='$'))
+
+
+def pri_ssh_key_file(file_name: str) -> str:
+    """
+    Wrapper for args.get_text_file_contents with appropriate arguments for SSH key files.
+    At this point, we just make sure that the files are at least 256 characters long.
+    """
+    logging.info(
+        f"Reading in SSH private key from '{file_name}' file")
+    return args.get_text_file_contents(
+        file_name=file_name,
+        value_validator=lambda s: args.validate_string(s, min_length=256))
+
+
+def pub_ssh_key_file(file_name: str) -> str:
+    """
+    Wrapper for args.get_text_file_contents with appropriate arguments for SSH key files.
+    At this point, we just make sure that the files are at least 256 characters long.
+    """
+    logging.info(
+        f"Reading in SSH public key from '{file_name}' file")
+    return args.get_text_file_contents(
+        file_name=file_name,
+        value_validator=lambda s: args.validate_string(s, min_length=256))
+
+
+def prompt_for_password() -> str:
+    """
+    Prompt the user to enter their desired password string, then
+    have them enter it again. If they do not match, repeat. If they do
+    match, return the password.
+    """
+    logging.debug("Prompting user for password")
+    while True:
+        pw_text = getpass.getpass("Enter desired password: ")
+        if pw_text == getpass.getpass("Verify password: "):
```

Comment:
Oh I guess one other question while I'm looking at this, do you want to check anywhere that the password they entered isn't just an empty string? Or any of the other validation stuff like you're doing for the env var case? 

Otherwise it's pretty easy to just hit enter twice to use the empty string as a password. Not sure, maybe that's fine.

Reply:
Done!